src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java

```diff
@@ -48,15 +46,11 @@ public void run() {
                 .stream()
                 .map(this::tryParse)
                 .filter(Objects::nonNull)
-                .forEach(parsedReport -> {
-                    List<Boolean> successStatuses = parsedReport
-                        .statuses
-                        .stream()
-                        .map(this::trySendUpdate)
-                        .collect(toList());
-
-                    if (successStatuses.stream().allMatch(s -> s == true)) {
-                        ftpClient.deleteReport(parsedReport.path);
+                .map(this::processReport)
+                .reduce(Boolean::logicalAnd)
+                .ifPresent(allSuccess -> {
+                    if (allSuccess) {
+                        sendLetterClient.checkPrintStatus();
```

Comment:
this should be always called I think

Reply:
would it not skew reports too much depending on job interval? at the moment it is half an hour

we would still push the exception/event to app insights regarding the failure of parsing individual row in the csv. print status check would additionally report more or less the same thing.